tests/test_distributions.py

```diff
@@ -333,6 +498,25 @@ def test_eq_ne_hash() -> None:
         assert not d != d_copy
         assert hash(d) == hash(d_copy)
 
+    # Different field values.
+    di0 = distributions.FloatDistribution(low=1, high=2)
+    di1 = distributions.FloatDistribution(low=1, high=3)
+    assert di0 != di1
+    assert not di0 == di1
+    assert hash(di0) != hash(di1)
+
+    # Different distribution classes.
+    di2 = distributions.IntDistribution(low=1, high=2)
+    assert di0 != di2
+    assert not di0 == di2
+    assert hash(di0) != hash(di2)
+
+    # Different types.
+    assert di0 != 1
```

Comment:
[question] Do we have to check a type here? It feels too much testing to me. 💭

Reply:
My random guess is that these checks for existing classes might have tried to check the second condition of `__eq__()`. However, the check seems to be trivial and other classes do not have such check, so I think fine to remove it.